Source/Fuse.Common/NameValuePair.uno

```diff
@@ -0,0 +1,56 @@
+using Uno;
+using Uno.Collections;
+
+namespace Fuse
+{
+	/** Represents a name-value pair, as denoted by `name: value` in UX expressions.
+		
+		Implements `IObject`, which means the `NameValuePair` can be viewed as an object
+		with a single property.
+
+		An `IArray` containing some `NameValuePairs` can be converted to an `IObject`
+		implementation containing all those properties using the `ObjectFromArray` method.
+	*/
+	public sealed class NameValuePair : IObject
+	{
+		public string Name { get; private set; }
+		public object Value { get; private set; }
+		public NameValuePair(string name, object value)
+		{
+			Name = name;
+			Value = value;
+		}
+
+		public override string ToString()
+		{
+			return "(" + Name + ": " + Value + ")";
+		}
+
+		string[] IObject.Keys { get { return new [] { Name }; } }
+		bool IObject.ContainsKey(string key) { return Name == key; } 
+		object IObject.this[string key] 
+		{
+			get
+			{
+				if (key != Name) throw new ArgumentException("Object (NameValuePair) does not contain the given key");
+				return Value;
+			}
+		}
+
+		/** Creates an IObject implementation from an `IArray` of `NameValuePair`.
+			The array doesn't need to contain exclusively `NameValuePair` instances. Objects of other
+			types are ignored.
+		*/
+		public static IObject ObjectFromArray(IArray list)
```

Comment:
It would be needed pretty soon. Any feature that actualy relies on NameValuePairs will probably want to convert an IArray of those to an IObject.